docs/user-manual/event-streaming.md

```diff
@@ -0,0 +1,298 @@
+
+The class **EventStreamer** is the main interface for streaming event and simply provide three virtual methods to implement :
+
+```cpp
+// factory method to create an event with the correct type
+virtual Event* createEvent() const = 0;
+// write an event
+virtual StatusCode write(const Event* const pEvent, xdrstream::IODevice* pDevice) = 0;
+// read an event
+virtual StatusCode read(Event* &pEvent, xdrstream::IODevice* pDevice) = 0;
+```
+
+All the streaming functionalities are implemented in the [xdrstream](https://github.com/DQM4HEP/xdrstream) package installed with DQM4hep packages while building the software. The class **IODevice** is the main interface for reading/writing raw data from/to a buffer. The different sub-classes of **IODevice** implement *how* the data are streamed, e.g in a buffer, in a file, etc ... Most of the time, the real implementation is a buffer.
+
+The best way to show how this whole mechanism works and also to show how this IODevice class works, is to provide a simple example. 
+
+Let say you have setup a device that provide measurement of temperature and pressure every seconds. We first have to define the event model for this setup. This is straight forward :
+
+```cpp
+// Defined in DeviceEvent.h
+
+class DeviceEvent {
+public:
+  float m_temperature = {0.f};
+  float m_pressure = {0.f};
+};
+```
+
+The second step is to implement the streamer. Let's start with the class definition:
+
+```cpp
+// DeviceEventStreamer.cc
+#include <dqm4hep/EventStreamer.h>
+#include <DeviceEvent.h> // our event definition
+
+using namespace dqm4hep::core;
+
+class DeviceEventStreamer : public EventStreamer {
+public:
+  ~DeviceEventStreamer() {}
+  Event* createEvent() const;
+  StatusCode write(const Event* const pEvent, xdrstream::IODevice* pDevice);
+  StatusCode read(Event* &pEvent, xdrstream::IODevice* pDevice);
+};
+
+// next is the implementation
+```
+
+We first include our event definition and the event streamer from DQM4hep. The class definition is really simple. Let's write the implementation, step by step.
+
+```cpp
+Event* DeviceEventStreamer::createEvent() const {
+  
+  DeviceEvent* devEvt = new DeviceEvent(); 
+  Event* event = new EventBase<DeviceEvent>(devEvt);
+  return event;
+}
+```
+
+The first line creates our event and the second wraps it into a DQM4hep event instance. Simple.
+
+```cpp
+StatusCode DeviceEventStreamer::write(const Event* const pEvent, xdrstream::IODevice* pDevice) {
+  
+  const DeviceEvent* devEvt = pEvent->getEvent<DeviceEvent>();
+
+  if(nullptr == devEvt)
+    return STATUS_CODE_INVALID_PARAMETER;
+  
+  // write all the properties of the DQM4hep event definition
+  if( ! XDR_TESTBIT( pEvent->writeBase( pDevice ) , xdrstream::XDR_SUCCESS ) )
+    return STATUS_CODE_FAILURE;
+    
+  // write temperature
+  if( ! XDR_TESTBIT( pDevice->write<float>( & devEvt->m_temperature ),  , xdrstream::XDR_SUCCESS ) )
```

Comment:
if( ! XDR_TESTBIT( pDevice->write<float>( & devEvt->m_temperature ),  xdrstream::XDR_SUCCESS ) )
You have one too many comma in the parameters